external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -2564,16 +2597,15 @@ fn sequence_item(context: &mut Context, sp!(loc, pitem_): P::SequenceItem) -> E:
 }
 
 fn exps(context: &mut Context, pes: Vec<P::Exp>) -> Vec<E::Exp> {
-    pes.into_iter().map(|pe| exp_(context, pe)).collect()
-}
-
-fn exp(context: &mut Context, pe: P::Exp) -> Box<E::Exp> {
-    Box::new(exp_(context, pe))
+    pes.into_iter()
+        .map(|pe| *exp(context, Box::new(pe)))
+        .collect()
 }
 
-fn exp_(context: &mut Context, sp!(loc, pe_): P::Exp) -> E::Exp {
```

Comment:
Nit: it feels like this change caused a lot of code churn in this file without a lot of gain to show for it. Is there a particular motivation?

Reply:
Some quick experimental analysis suggests a pretty significant win (taking on average 85% memory over previous for expansion, probably even better for naming). Nice!